api/v1alpha1/nutanix_node_types.go

```diff
@@ -0,0 +1,286 @@
+	NutanixBootTypeLegacy NutanixBootType = "legacy"
+
+	// NutanixBootTypeUEFI is a resource identifier identifying the UEFI boot type for virtual machines.
+	NutanixBootTypeUEFI NutanixBootType = "uefi"
+)
+
+// NutanixIdentifierType is an enumeration of different resource identifier types.
+type NutanixIdentifierType string
+
+func (NutanixIdentifierType) VariableSchema() clusterv1.VariableSchema {
+	return clusterv1.VariableSchema{
+		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+			Type:        "string",
+			Description: "NutanixIdentifierType is an enumeration of different resource identifier types",
+			Enum: variables.MustMarshalValuesToEnumJSON(
+				NutanixIdentifierName,
+				NutanixIdentifierUUID,
+			),
+		},
+	}
+}
+
+// NutanixBootType is an enumeration of different boot types.
+type NutanixBootType string
+
+func (NutanixBootType) VariableSchema() clusterv1.VariableSchema {
+	return clusterv1.VariableSchema{
+		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+			Type:        "string",
+			Description: "NutanixBootType is an enumeration of different boot types.",
+			Enum: variables.MustMarshalValuesToEnumJSON(
+				NutanixBootTypeLegacy,
+				NutanixBootTypeUEFI,
+			),
+		},
+	}
+}
+
+// NutanixGPUIdentifierType is an enumeration of different resource identifier types for GPU entities.
+type NutanixGPUIdentifierType string
+
+func (NutanixGPUIdentifierType) VariableSchema() clusterv1.VariableSchema {
+	return clusterv1.VariableSchema{
+		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+			Type:        "string",
+			Description: "NutanixGPUIdentifierType is an enumeration of different resource identifier types for GPU entities.",
+			// TODO:deepakm-ntnx add GPU Identifier type enums
+			// Enum: variables.MustMarshalValuesToEnumJSON(
+			// ),
+		},
+	}
+}
+
+type NutanixMachineDetails struct {
+	// vcpusPerSocket is the number of vCPUs per socket of the VM
+	VCPUsPerSocket int32 `json:"vcpusPerSocket"`
+
+	// vcpuSockets is the number of vCPU sockets of the VM
+	VCPUSockets int32 `json:"vcpuSockets"`
+
+	// memorySize is the memory size (in Quantity format) of the VM
+	MemorySize string `json:"memorySize"`
+
+	// image is to identify the rhcos image uploaded to the Prism Central (PC)
+	// The image identifier (uuid or name) can be obtained from the Prism Central console
+	// or using the prism_central API.
+	Image NutanixResourceIdentifier `json:"image"`
+
+	// cluster is to identify the cluster (the Prism Element under management
+	// of the Prism Central), in which the Machine's VM will be created.
+	// The cluster identifier (uuid or name) can be obtained from the Prism Central console
+	// or using the prism_central API.
+	Cluster NutanixResourceIdentifier `json:"cluster"`
+
+	// subnet is to identify the cluster's network subnet to use for the Machine's VM
+	// The cluster identifier (uuid or name) can be obtained from the Prism Central console
+	// or using the prism_central API.
+	Subnets NutanixResourceIdentifiers `json:"subnet"`
```

Comment:
This is of type `NutanixResourceIdentifiers` which is already a slice but the `json` representation is `subnet, should this be a single subnet or a list?